### ISSUE TITLE
Solr Search core indexing Authority MetadataValue fails

### ISSUE THREAD

**Issue by Raf-atmire** [DSpace/DSpace#3193]
**Describe the bug**
Currently it's not possible in DSpace 7 to index an authority value into Solr for a Metadata Field which doesn't have either a vocabulary or a pairs type (excluding qualdrop). The indexing of an Item containing such a MetadataValue will produce a NullPointerException and cause the entire Item to not be indexed into Solr.
The code responsible for the NullPointerException can be found <a href="https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java#L504">here</a>. It'll try to find an entry in the `controllerFormDefinitions` but it won't have an entry and thus provide a NullPointer on the subsequent `.get(NAME)` call. This `controllerFormDefinitions` map is constructed <a href="https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java#L425">here</a> which is called <a href="https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java#L368">here</a>. This produces a map of metadatafields for which there exists an entry in the item-submission.xml that has either a pairsType (excluding qualdrop inputTypes) or a vocabulary. If neither exists, the metadataField won't be added to this map and the above mentioned Nullpointer will occur.
<a href="https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java#L496">this</a> method is called from a lot of different places, but for the Solr indexing, it happens <a href="https://github.com/DSpace/DSpace/blob/2b4f22be65d9997194d30cf01a49b9cac539b4a2/dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java#L194">here</a> which is in turn called by the <a href="https://github.com/DSpace/DSpace/blob/2b4f22be65d9997194d30cf01a49b9cac539b4a2/dspace-api/src/main/java/org/dspace/discovery/indexobject/ItemIndexFactoryImpl.java#L364">ItemIndexFactory</a> and the <a href="https://github.com/DSpace/DSpace/blob/2b4f22be65d9997194d30cf01a49b9cac539b4a2/dspace-api/src/main/java/org/dspace/discovery/SolrServiceMetadataBrowseIndexingPlugin.java#L177">SolrServiceMetadataBrowseIndexingPlugin</a>. Both of these will crash if the metadataField which contains a MetadataValue holding an Authority is present within the Item if that MetadataField doesn't have the above mentioned config in the item-submission, resulting in said Item being unable to be indexed.

**To Reproduce**
1. Enable config for the Solr Authority indexing for a metadata field
```
choices.plugin.dc.contributor.author = SolrAuthorAuthority
authority.controlled.dc.contributor.author = true
```
2. Create an Item with an AuthorityValue in a dc.contributor.author metadataValue
3. Proceed to try to search for this value in Discovery
4. Verify that the Item doesn't show up, even after reindexing

**Expected behavior**
The Item should be present in the Solr Search core

**Related work**
This bug was discovered through the development of <a href="https://github.com/DSpace/DSpace/pull/3176">this</a> PR whilst trying to create the test `discoverFacetsAuthorWithAuthorityWithSizeParameter` in `DiscoveryRestControllerIT`

**Comment by tdonohue** [comment]
@Raf-atmire : As it sounds like you've done a lot of digging on the problem here, is this ticket of interest to you or your team?

**Comment by Raf-atmire** [comment]
@tdonohue This is definitely of interest to us, we'd like to work on this

**Comment by tdonohue** [comment]
@Raf-atmire : Sounds good, please provide an estimate & we'll get this scheduled.  Assuming it's a small fix, we may still be able to fit it into Beta 5....but if not, we can definitely include it in time for 7.0 final.

**Comment by tdonohue** [comment]
I'm moving this ticket into the 7.0 board, and flagging as `medium priority`.  @Raf-atmire assigning to you to analyze and provide feedback on the best fix.   As noted, assuming this is small (hoping it will be), I'd like to get this into the 7.0 release.

**Comment by abollini** [comment]
is this different than https://github.com/DSpace/DSpace/issues/3177  / https://github.com/DSpace/DSpace/pull/3196 ?

**Comment by tdonohue** [comment]
Good catch @abollini : This might be a duplicate & already be solved by #3196.

@Raf-atmire : Could you retest this? If it's solved, let's close it out.  If there's still an issue remaining, then we'll need an estimate, etc.

**Comment by tdonohue** [comment]
Reassigning to @benbosman (per today's meeting).  @benbosman : this needs analysis on whether it's already been fixed (possibly by #3196) or if additional issues still exist (see above)

**Comment by Raf-atmire** [comment]
It wasn't fixed yet in the current main branch, I've tested this by removing the methods we've used to circumvent this problem in the initial PR for the `discoverFacetsAuthorWithAuthorityWithSizeParameter` in `DiscoveryRestControllerIT`

We'll be working around this problem by providing a try-catch around the classes that use the [getLabel](https://github.com/DSpace/DSpace/blob/2b4f22be65d9997194d30cf01a49b9cac539b4a2/dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java#L194) method. We'll catch any error thrown and provide a log.warn instead so that the index-discovery doesn't crash but instead logs that it couldn't find the label.

I estimate this to be 4 hours of work to provide the try-catches in all the necessary places and to properly test and reconfigure the IT for this.
